pkg/apis/k0s/v1beta1/elb.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2024 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+// Defaults are keepalived's defaults.
+const defaultVirtualRouterID = 51
+const defaultAdvertInterval = 1
+const defaultInterface = "eth0"
+
+// ExternalLoadBalancingSpec defines the configuration options related to k0s's
+// keepalived feature.
+type ExternalLoadBalancingSpec struct {
+	VRRPInstances VRRPInstances `json:"vrrpInstances,omitempty"`
```

Comment:
I really like this API for configuring VRRP instances, if you check how VRRP is done outside keepalived, this API would  without any change to other VRRP software [such as cisco IOS](https://www.cisco.com/c/en/us/td/docs/routers/crs/software/crs_r4-0/addr_serv/configuration/guide/ic40crs1book_chapter10.html).  However, if we need to switch later to something that does not use VRRP?

I don't see a really good solution for this. I considered using `leaderElection` instead of `VRRPInstances`, however we **REALLY** need to be able to define the VRID and Authpass.

I would love a better name if someone can think of something.